modules/programs/waybar.nix

```diff
@@ -0,0 +1,371 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with lib.types;
+let
+  cfg = config.programs.waybar;
+
+  # Used when generating warnings
+  modulesPath = "programs.waybar.settings.modules";
+
+  # Taken from <https://github.com/Alexays/Waybar/blob/adaf84304865e143e4e83984aaea6f6a7c9d4d96/src/factory.cpp>
+  defaultModuleNames = [
```

Comment:
The `battery` module is missing from this list.

I get the following error message when I had it included under `settings/0/modules/` which is a little misleading:

> error: The option value `home-manager.users.wkral.programs.waybar.settings.[definition 1-entry 1].modules' in `/home/wkral/home-nix/config/waybar' is not of type `attribute set'.

Reply:
This error is unrelated to the generation of warnings (warnings != error).
This error means your definition for a module in `modules` is not a set.

Reply:
I'll try and be more clear here, this isn't related to warnings. Because `battery` isn't in this list it fails the `isValidModule` check. Using `lib.types.addCheck` `isValidModule` is added to the type checks for `attrs` for the `modules` option. 

So by having a module name not in this list and not prefixed with `custom/` I get this error message telling me that the modules attribute set is not an attribute set which I found rather confusing.

You can see the config generating this error, though it's currently commented out: https://github.com/wkral/home-nix/blob/8264ba6dbc31e3b9a1d25f46fbbbf55a1661a9d9/config/waybar/default.nix#L109

Reply:
Thanks for the feedback. I changed invalid custom module names into warnings in commit 1000ea5